base/special/exp.jl

```diff
@@ -198,69 +198,98 @@ end
 #
 # 2. Approximate b^r by 7th-degree minimax polynomial p_b(r) on the interval [-log(b,2)/2, log(b,2)/2].
 # 3. Scale back: b^x = 2^N * p_b(r)
-
 # For both, a little extra care needs to be taken if b^r is subnormal.
 # The solution is to do the scaling back in 2 steps as just messing with the exponent wouldn't work.
-for (func, base) in (:exp2=>Val(2), :exp=>Val(:ℯ), :exp10=>Val(10))
-    @eval begin
-        function ($func)(x::T) where T<:Float64
-            N_float = muladd(x, LogBo256INV($base, T), MAGIC_ROUND_CONST(T))
-            N = reinterpret(UInt64, N_float) % Int32
-            N_float -=  MAGIC_ROUND_CONST(T) #N_float now equals round(x*LogBo256INV($base, T))
-            r = muladd(N_float, LogBo256U($base, T), x)
-            r = muladd(N_float, LogBo256L($base, T), r)
-            k = N >> 8
-            jU, jL = table_unpack(N&255 +1)
-            small_part =  muladd(jU, expm1b_kernel($base, r), jL) + jU
-
-            if !(abs(x) <= SUBNORM_EXP($base, T))
-                x >= MAX_EXP($base, T) && return Inf
-                x <= MIN_EXP($base, T) && return 0.0
-                if k <= -53
-                    # The UInt64 forces promotion. (Only matters for 32 bit systems.)
-                    twopk = (k + UInt64(53)) << 52
-                    return reinterpret(T, twopk + reinterpret(UInt64, small_part))*(2.0^-53)
-                end
-            end
-            twopk = Int64(k) << 52
-            return reinterpret(T, twopk + reinterpret(Int64, small_part))
-        end
 
-        function ($func)(x::T) where T<:Float32
-            N_float = round(x*LogBINV($base, T))
-            N = unsafe_trunc(Int32, N_float)
-            r = muladd(N_float, LogBU($base, T), x)
-            r = muladd(N_float, LogBL($base, T), r)
-            small_part = expb_kernel($base, r)
-            if !(abs(x) <= SUBNORM_EXP($base, T))
-                x > MAX_EXP($base, T) && return Inf32
-                x < MIN_EXP($base, T) && return 0.0f0
-                if N<=Int32(-24)
-                    twopk = reinterpret(T, (N+Int32(151)) << Int32(23))
-                    return (twopk*small_part)*(2f0^(-24))
-                end
-                N == (exponent_max(T)+1) && return small_part * T(2.0) * T(2.0)^exponent_max(T)
-            end
-            twopk = reinterpret(T, (N+Int32(127)) << Int32(23))
-            return twopk*small_part
+@inline function exp_impl(x::T, base) where T<:Float64
```

Comment:
```suggestion
@inline function exp_impl(x::Float64, base)
```

what does the `_imp` postfix mean in these methods?

Reply:
Short for implimentation. It's there a better name to use?

Reply:
ah right... Yeah we could keep it as is or use an underscore prefix since it's a 'private' method.